cuda/insert.cpp

```diff
@@ -0,0 +1,39 @@
+#include <bits/stdc++.h>
+#include <cuda_runtime.h>
+
+#include "kernels.h"
+
+using namespace std;
+
+void insert_point(Point new_point, Grid *root_grid,
+				  vector<QuadrantBoundary> &boundaries,
+				  unordered_map<int, Grid *> &grid_map, int quadrant_id) {
+	// Access the target grid from the unordered map
+	Grid *target_grid = grid_map[quadrant_id];
+
+	// Add the point to the grid
+	Point *new_points =
+		(Point *)malloc((target_grid->count + 1) * sizeof(Point));
+	memcpy(new_points, target_grid->points, target_grid->count * sizeof(Point));
+	new_points[target_grid->count] = new_point;
```

Comment:
first off we can put target_grid->count in a variable since its being accessed multiple times. Also, here it should be target_grid->count+1 . right?